Masa.Utils.Data.Promethus/Extensions/ObjectExtensions.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+using System.Collections;
+using System.Reflection;
+using System.Text;
+using System.Web;
+
+namespace Masa.Utils.Data.Promethus;
+
+public static class ObjectExtensions
+{
+    /// <summary>
+    /// not support System.text.json
+    /// </summary>
+    /// <param name="obj"></param>
+    /// <param name="isEnumString"></param>
+    /// <param name="isCamelCase"></param>
+    /// <param name="isUrlEncode"></param>
+    /// <returns></returns>
+    public static string? ToUrlParam(this object obj, bool isEnumString = true, bool isCamelCase = true, bool isUrlEncode = true)
+    {
+        return GetValue(obj, string.Empty, isEnumString, isCamelCase, isUrlEncode);
+    }
+
+    private static string? GetValue(object obj, string preStr, bool isEnumString = false, bool isCamelCase = true, bool isUrlEncode = true)
+    {
+        if (obj == null) return null;
+        var type = obj.GetType();
+        if (type == typeof(string))//string
+        {
+            var str = (string)obj;
+            return AppendValue(preStr, str, "=", isUrlEncode);
+        }
+        else if (type.IsValueType)
+        {
+            if (type.IsEnum)//enum
+            {
+                var str = isEnumString ? obj.ToString() : Convert.ToInt32(obj).ToString();
+                return AppendValue(preStr, str, "=", isUrlEncode);
+            }
+            else if (!type.IsPrimitive) //struct
+            {
+                return GetObjValue(type, obj, preStr, isEnumString, isCamelCase, isUrlEncode);
+            }
+            else //sample value
+            {
+                var str = obj.ToString();
+                return AppendValue(preStr, str, "=", isUrlEncode);
+            }
+        }
+        else if (type.IsArray || type.GetInterfaces().Any(t => t.Name.IndexOf("IEnumerable") == 0))
+        {
+            return GetEnumerableValue(obj, preStr, isEnumString, isCamelCase, isUrlEncode);
+        }
+        else if (type.IsClass)
+        {
+            return GetObjValue(type, obj, preStr, isEnumString, isCamelCase, isUrlEncode);
+        }
+        else
+        {
+            //
+            return null;
+        }
+    }
+
+    private static string GetObjValue(Type type, object obj, string preStr, bool isEnumString = false, bool isCamelCase = true, bool isUrlEncode = true)
```

Comment:
preStr?

Reply:
Nested objects used, like  this:
```
  {
     user : {
          name: "name",
          age: 20    
    }
  }
```
Convert process: user.name=name, user.age=20 ,"user." is ```preStr```